test/test_datasets.py

```diff
@@ -480,6 +460,48 @@ def inject_fake_data(self, tmpdir, config):
         return num_images_per_category * len(categories)
 
 
+@unittest.skipIf(sys.platform in ('win32', 'cygwin'), 'temporarily disabled on Windows')
+class WIDERFaceTestCase(datasets_utils.ImageDatasetTestCase):
+    DATASET_CLASS = datasets.WIDERFace
+    FEATURE_TYPES = (PIL.Image.Image, (dict, type(None)))  # test split returns None as target
+    CONFIGS = datasets_utils.combinations_grid(split=('train', 'val', 'test'))
+
+    def inject_fake_data(self, tmpdir, config):
+        widerface_dir = pathlib.Path(tmpdir) / 'widerface'
+        annotations_dir = widerface_dir / 'wider_face_split'
+        os.makedirs(annotations_dir)
+
+        for split in ('train', 'val', 'test'):
+            split_idx = {
+                'train': 1,
+                'val': 2,
+                'test': 3,
+            }[split]
+
+            num_examples = 1
```

Comment:
Instead of hard-coding this you could do something like `num_examples = split_idx`. That way we can be sure that the dataset is loading the correct data given by `split`. If all splits have the same length, we can't use that as indicator.